zebra/label_manager.c

```diff
@@ -69,49 +68,72 @@ static int relay_response_back(struct zserv *zserv)
 	uint8_t version;
 	vrf_id_t vrf_id;
 	uint16_t resp_cmd;
+	uint8_t proto;
+	const char *proto_str;
+	unsigned short instance;
+	struct zserv *zserv;
 
+	/* input buffer with msg from label manager */
 	src = zclient->ibuf;
 	dst = obuf;
 
 	stream_reset(src);
 
+	/* parse header */
 	ret = zclient_read_header(src, zclient->sock, &size, &marker, &version,
 				  &vrf_id, &resp_cmd);
 	if (ret < 0 && errno != EAGAIN) {
 		zlog_err("%s: Error reading Label Manager response: %s",
 			 __func__, strerror(errno));
 		return -1;
 	}
-	zlog_debug("%s: Label Manager response received, %d bytes", __func__,
-		   size);
+	zlog_debug("Label Manager response received, %d bytes", size);
 	if (size == 0)
 		return -1;
 
-	/* send response back */
+	/* Get the 'proto' field of the message */
+	proto = stream_getc(src);
+
+	/* Get the 'instance' field of the message */
+	instance = stream_getw(src);
+
+	proto_str = zebra_route_string(proto);
+
+	/* lookup the client to relay the msg to */
+	zserv = zebra_find_client(proto, instance);
+	if (!zserv) {
+		zlog_err("Error sending LM response back: can't find client with proto %s, instance %u",
+				proto_str, instance);
+		return -1;
+	}
+	zlog_debug("Found client to relay LM msg to: %s instance %u",
+				proto_str, instance);
+
+	/* copy msg into output buffer */
+	dst = obuf;
 	stream_copy(dst, src);
-	ret = writen(zserv->sock, src->data, stream_get_endp(src));
+
+	/* send response back */
+	ret = writen(zserv->sock, dst->data, stream_get_endp(dst));
 	if (ret <= 0) {
-		zlog_err("%s: Error sending Label Manager response back: %s",
-			 __func__, strerror(errno));
+		zlog_err("Error relaying Label Manager response to %s instance %u: %s",
+				proto_str, instance, strerror(errno));
```

Comment:
The style checker is not complaining, but I think the indentation is wrong here and in several other places. I think it would be a good idea to run clang-format to indent your changes.

Reply:
Also, please fix the new SA warning.

Reply:
Yes, the code combines the daemon protocol and instance Id to form a unique identifier.
This was actually the way it worked before these patches. The problem was that the relay/proxy function was not properly done.

In order to disambiguate requests from distinct routers, users willing to use the relay/proxy functionality (i.e. an external label manager) must handle the uniqueness by providing the required instance Id for each daemon. In **LDPd**, this can be set with the **'-n'** cmd line argument. For BGPd, tomorrow we will send a patch that allows setting (and propagating) a certain instance Id from the command line argument similarly.

Reply:
I have resubmitted the patches and:
  1) fixed the title of the second commit as suggested.
  2) fixed the SA warning we introduced.
  3) fixed (and unified the style of) some logs to get rid of formatting warnings.

The *checkpatch.pl* script does not issue warnings anymore.